components/style/matching.rs

```diff
@@ -383,7 +349,32 @@ pub fn rare_style_affecting_attributes() -> [Atom; 3] {
     [ atom!("bgcolor"), atom!("border"), atom!("colspan") ]
 }
 
-static STYLE_SHARING_CANDIDATE_CACHE_SIZE: usize = 40;
+fn have_same_class<E: TElement>(element: &E, candidate: &E) -> bool {
+    // XXX Efficiency here, I'm only validating ideas.
+    let mut first = vec![];
+    let mut second = vec![];
```

Comment:
These could be sets to be more accurate (fewer misses). If on the other hand we don’t mind being conservative (more misses) comparing the string values of `class` attributes might be slightly faster than building these vecs.